calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -0,0 +1,324 @@
+        expected_return = '15'
+        m_client.get_default_node_as.return_value = expected_return
+
+        # Call method under test
+        show_default_node_as()
+
+        # Assert
+        m_client.get_default_node_as.assert_called_once_with()
+        self.assertEqual(m_stdout.getvalue().strip(),expected_return)
+
+    @patch('calico_ctl.bgp.client',autospec=True)
+    @patch('sys.stdout',new_callable=StringIO)
+    def test_show_bgp_node_mesh(self,m_stdout,m_client):
+        """
+        Test for show_bgp_node_mesh() for calicoctl bgp
+        """
+        # Set up mock objects
+        expected_return = '15'
+        m_client.get_bgp_node_mesh.return_value = expected_return
+
+        # Call method under test
+        show_bgp_node_mesh()
+
+        # Assert
+        m_client.get_bgp_node_mesh.assert_called_once_with()
+        self.assertEqual(m_stdout.getvalue().strip(),'on')
+
+    @patch('calico_ctl.bgp.client',autospec=True)
+    @patch('sys.stdout',new_callable=StringIO)
+    def test_show_bgp_node_mesh_fail(self,m_stdout,m_client):
+        """
+        Test for show_bgp_node_mesh() for calicoctl bgp
+        """
+        # Set up mock objects
+        expected_return = ''
+        m_client.get_bgp_node_mesh.return_value = expected_return
+
+        # Call method under test
+        show_bgp_node_mesh()
+
+        # Assert
+        m_client.get_bgp_node_mesh.assert_called_once_with()
+        self.assertEqual(m_stdout.getvalue().strip(),'off')
+
+    @patch('calico_ctl.bgp.client',autospec=True)
+    def test_set_bgp_node_mesh(self,m_client):
+        """
+        Test for set_bgp_node_mesh for calicoctl bgp
+        """
+        # Call method under test
+        set_bgp_node_mesh(True)
+
+        # Assert
+        m_client.set_bgp_node_mesh.assert_called_once_with(True)
+
+
+
+class TestCheckSystem(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+
+class TestContainer(unittest.TestCase):
+
+    @patch('sys.exit',autospec=True)
+    def test_validate_arguments(self, m_sys_exit):
+        """
+        Test validate_arguments for calicoctl container command
+        """
+        # Set up arguments
+        cases = (({'<CONTAINER>':'node1', 'ip':1, 'add':1, '<IP>':'127.a.0.1'},False),
+                 ({'<CONTAINER>':'node1', 'ip':1, 'add':1, '<IP>':'aa:bb::zz'},False) ,
+                 ({'add':1, '<CONTAINER>':'node1', '<IP>':'127.a.0.1'},False) ,
+                 ({'add':1, '<CONTAINER>':'node1', '<IP>':'aa:bb::zz'},False))
+
+        # Call method under test for each test case
+        # Assert that method exits on bad input
+        for case, is_valid in cases:
+            print "Testing case %s ..." % case
+            container_validate_arguments(case)
+            if is_valid:
+                assert not m_sys_exit.called
+            else:
+                assert m_sys_exit.called
+            m_sys_exit.reset_mock()
+
+
+class TestDiags(unittest.TestCase):
+
+    def setUp(self):
+        pass
```

Comment:
This class can be removed, I don't think there are any functions that need to be tested within this.